src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -1714,13 +1714,20 @@ private static async Task<IList<CompatibilityCheckResult>> VerifyCompatibilityAs
 
             Tuple<bool, List<RestoreTargetGraph>, RuntimeGraph> result = null;
             bool failed = false;
+
+            Dictionary<NuGetFramework, string> targetFrameworkToAlias = projectFrameworkRuntimePairs.Where(e => string.IsNullOrEmpty(e.RuntimeIdentifier)).ToDictionary(
+                        p => p.Framework,
+                        p => p.TargetAlias,
+                        NuGetFramework.Comparer);
+
             using (telemetryActivity.StartIndependentInterval(CreateRestoreTargetGraphDuration))
             {
                 try
                 {
                     result = await projectRestoreCommand.TryRestoreAsync(
                         projectRange,
-                        projectFrameworkRuntimePairs,
+                        projectFrameworkRuntimePairs.Select(e => new FrameworkRuntimePair(e.Framework, e.RuntimeIdentifier)).ToList(),
```

Comment:
perf nitpick: there's opportunity to remove a few more theoretical allocations here as well. Rather than LINQ's ToList, crate a list with the correct capacity, and use a loop that doesn't allocate an enumerator or delegate.

But it looks like it should be once per target framework per project, so maybe this isn't yet nested in enough loops to be concerned about.